### PR TITLE
Php 8 fix

### PR DESCRIPTION
**Pull request by jcheron**
### Fixed
- namespace parsing in `AnnotationParser->parse` with php 8 (see [namespaced names as single token RFC](https://wiki.php.net/rfc/namespaced_names_as_token))

### Notes
- php 8 required

### CODE REVIEW

**Comment by mindplay-dk** [comment]
PHP 8 has [attributes](https://www.php.net/manual/en/language.attributes.overview.php).

It's questionable whether projects targeting PHP 8 need this library at all - probably what you should do, is port your annotations to native PHP 8 attributes and stop using this library altogether.

You're proposing we drop compatibility with PHP 7. Merging/releasing this change would preclude fixing any bugs for PHP 7 users in the future - so that's probably a no-go, since PHP 7 is where this library is needed, since it *doesn't* have attributes.

I would accept a PR that introduces compatibility with PHP 8 and maintains backwards compatibility - since this would help people migrate from this library to native PHP 8 attributes. I don't know if that's possible.

Alternatively, you could publish your package as a replacement package for migration purposes.

Making this package support PHP 8 isn't very interesting for any purposes besides migration - most likely, any future interest in this package will be for legacy projects only.

**Comment by jcheron** [comment]
Hi @mindplay-dk 

I don't use `php-annotations` for my own needs, but as part of a framework.
In this context, annotations are used by end users of this framework, to define multiple elements of their application in an MVC architecture.

It is possible that some of these users may wish to migrate their projects to php 8.
It would be unfortunate if they were obliged to rewrite all the annotations used as php8 attributes in order to do so.

In this framework, annotations are never used at runtime. So I don't have concerns about performance, but more about the comfort of writing annotations for the user, and the effortless migration possibilities for them.

In the near future, I think I will leave the choice to the users to use the old annotations, or the native php 8 attributes. But nothing is decided at this stage.
In the meantime, I am looking for a solution allowing them to migrate to PHP8, without constraining them to modify their code.

I can modify my PR to ensure backward compatibility with previous versions of PHP, I consider as you do that this is a requirement. 

I can also make a package migration to php8 from this project.
It's up to you, I don't care.

**Comment by mindplay-dk** [comment]
> I can modify my PR to ensure backward compatibility with previous versions of PHP, I consider as you do that this is a requirement.

That would be ideal.

It looks like `5.4` and `5.5` aren't supported by the default distro on Travis anymore - probably needs to be moved to the `matrix` section, similar to how `5.3` is configured, to get a working build.

Also, PHP 8 should not be an allowed failure - it looks like there are problems with the code coverage package. Probably the easiest path here would be to disable code-coverage for PHP 8.

**Comment by jcheron** [comment]
@mindplay-dk 
Ready for review

**Comment by mindplay-dk** [comment]
Please follow the [code-style](https://github.com/php-annotations/php-annotations#code-style) of the project: space after `if`-statements. The project uses 4 spaces for indentation - some files have mixed tabs and spaces now. (we probably ought to check in an `.editorconfig` file...)

Other than that it's looking good to go :-)

**Comment by jcheron** [comment]
Hi @mindplay-dk 
Is it good?

File: test/lib/xTestRunner.php [hunk 1]
```diff
@@ -55,14 +60,17 @@ public function __construct($rootPath, ResultPrinter $resultPrinter)
...
         $this->rootPath = $rootPath;
         $this->resultPrinter = $resultPrinter;
-
-        try {
-            $this->coverage = new \PHP_CodeCoverage();
-            $this->coverage->filter()->addDirectoryToWhitelist($rootPath);
-        } catch (\PHP_CodeCoverage_Exception $e) {
-            // can't collect coverage
+        if ($this->hasCoverage) {
+	        try {
```
Comment (mindplay-dk) [inline test/lib/xTestRunner.php:68]: There are still tabs in here - maybe do a search/replace?

**Comment by jcheron** [comment]
I had voluntarily disabled any formatting in my IDE, and any PSR fixer,
the choice of PSR2 formatter modifying certain parts of your code in an inappropriate manner .

So I format by hand, and the use of tabs has not been automatically replaced by spaces, sorry!

My apologies for this unforgivable negligence.
It was well worth this 10 days of penalties.

**Comment by jcheron** [comment]
Hi @mindplay-dk ,
I hope you are well, and that you just took some long holidays.

**Comment by jguillaumesio** [comment]
@mindplay-dk Maybe you could accept this PR, as a Ubiquity user we needs it and you have nothing to do more than merge it, @jcheron has done everything you asked for and it's a bit direspectful not to merge it now.

**Comment by mindplay-dk** [comment]
Sorry for the long response time.

For personal (health) reasons I am not active on GitHub or checking email regularly.

**Pull request merged**